spark/v3.4/spark-extensions/src/main/scala/org/apache/spark/sql/execution/datasources/v2/UpdateRowsExec.scala

```diff
@@ -51,6 +58,59 @@ case class UpdateRowsExec(
     copy(child = newChild)
   }
 
+  override def usedInputs: AttributeSet = {
```

Comment:
This is a similar optimization as in `ProjectExec`. It ensures if the same input attribute is used in multiple output expressions, we only fetch it once. Suppose I have `c1 = id + 10, c2 = id - 10`. Instead of fetching `id` for each output expression, we can do that once and reuse the same variable.